src/cdk/jsii-project.ts

```diff
@@ -424,7 +424,13 @@ export class JsiiProject extends TypeScriptProject {
     this.npmignore?.include(".jsii");
 
     if (options.docgen ?? true) {
-      new JsiiDocgen(this, { filePath: options.docgenFilePath });
+      // If jsiiVersion is "*", don't specify anything so the user can manage.
+      // Otherwise use a version that is compatible with all supported jsii releases.
+      const docgenVersion = options.jsiiVersion === "*" ? "*" : "^10.4.0";
```

Comment:
Will we have to upgrade this every time we add new TS support to docgen?

```suggestion
      const docgenVersion = options.jsiiVersion === "*" ? "*" : "^10.5.0";
```

Reply:
That really depends.

In general, yes. When a component decides to set a version constraint, it takes on the responsibility to maintain it. We do that in a large number of places in projen. 

But this specific constraint is wide enough to install the latest version of docgen anyway. It will probably work without intervention.

In practice, projen has a PR to update jsii stuff anyway when a new jsii version is released. So this is really only a little extra.